x/market/internal/keeper/swap.go

```diff
@@ -70,18 +70,22 @@ func (k Keeper) ComputeSwap(ctx sdk.Context, offerCoin sdk.Coin, askDenom string
 	// Apply only tobin tax without constant product spread
 	if offerCoin.Denom != core.MicroLunaDenom && askDenom != core.MicroLunaDenom {
 		spread = k.TobinTax(ctx)
-		illiquidTobinTaxList := k.IlliquidTobinTaxList(ctx)
-
-		// Apply highest tobin tax for the denoms in the swap operation
-		for _, tobinTax := range illiquidTobinTaxList {
-			if tobinTax.Denom == offerCoin.Denom ||
-				tobinTax.Denom == askDenom {
-				if tobinTax.TaxRate.GT(spread) {
-					spread = tobinTax.TaxRate
-				}
-			}
+		illiquidFactor, err2 := k.oracleKeeper.GetIlliquidFactor(ctx, offerCoin.Denom)
+		if err2 != nil {
+			return sdk.DecCoin{}, sdk.Dec{}, err2
 		}
 
+		askIlliquidFactor, err2 := k.oracleKeeper.GetIlliquidFactor(ctx, askDenom)
+		if err2 != nil {
+			return sdk.DecCoin{}, sdk.Dec{}, err2
+		}
+
+		// Apply highest illiquid factor for the denoms in the swap operation
+		if askIlliquidFactor.GT(illiquidFactor) {
+			illiquidFactor = askIlliquidFactor
+		}
+
+		spread = spread.Mul(illiquidFactor)
```

Comment:
Think it should be better data structure design to have unique tobin taxes registered for each of the whitelisted denoms instead of having a universal tobin tax and an "illiquid factor" for each denom.

Reply:
Resolved